plugins/two-factor-auth/frontend/app.js

```diff
@@ -194,7 +195,7 @@ function generateQRCode(username, secret, callback) {
                     // else everything is alright (login flow second phase)
                     else {
                         try {
-                            var verified = GA.check(req.body.auth_code, (member.two_factor_auth && member.two_factor_auth.secret_token) || req.body.secret_token);
+                            var verified = GA.check(req.body.auth_code, apiUtils.decrypt(member.two_factor_auth && member.two_factor_auth.secret_token || req.body.secret_token));
```

Comment:
So are we decrypting` req.body.secret_token` before sending to GA.check?

Reply:
We know it'll never be encrypted so we are using the fact that utils.decrypt doesn't try to decrypt stuff that don't have "[CLY_true]" at the end to avoid a longer ternary expression there :smile: 

https://github.com/Countly/countly-server/blob/master/api/utils/utils.js#L63

Reply:
Or what we want is this:
`GA.check(req.body.auth_code, member.two_factor_auth && member.two_factor_auth.secret_token ? apiUtils.decrypt(member.two_factor_auth.secret_token) : req.body.secret_token);`

Reply:
Yup, that's the longer ternary expression I wanted to avoid.

Reply:
I agree that it's hard to see why it's doing what it's doing without checking out utils.member. I'll use the ternary expression :smile:

Reply:
That "[CLY_true]" makes me nervous :) but since things are setup that way I got your point.